server/channels/store/sqlstore/post_store.go

```diff
@@ -2029,27 +2029,41 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 			excludedTerms = wildcard.ReplaceAllLiteralString(excludedTerms, ":* ")
 		}
 
-		excludeClause := ""
-		if excludedTerms != "" {
-			excludeClause = " & !(" + strings.Join(strings.Fields(excludedTerms), " | ") + ")"
-		}
+		// Regex to get quoted strings
+		quotedStringsReg := regexp.MustCompile(`("[^"]*")`)
```

Comment:
Would you consider adding this to the top of the file, it will still be compiled whenever the function is called. There is no need to compile it more than once.